.pipelines/azure_pipeline_validation_appmonitoring.yaml

```diff
@@ -0,0 +1,245 @@
+      inlineScript: |
+        cd appmonitoring/helm
+        pwd
+        az account set --subscription 66010356-d8a5-42d3-8593-6aaa3aeb1c11
+        az aks get-credentials --resource-group appmonitoring-pipeline-test --name aks-pipeline-testbed-ai
+        kubectl delete -f appmonitoring-deployment.yaml
+
+        echo "Wait for 10s for everything to clear up..."
+        sleep 10
+
+        export AIDEV_IMAGE_NAME=${{ variables.fullImageName }}
+        cat appmonitoring-deployment.yaml | envsubst | kubectl apply -f -
+
+        echo "Wait for 10s for charts to be applied..."
+        sleep 10
+
+  - task: AzureCLI@2
+    displayName: "Validate charts deployed to AKS"
+    inputs:
+      azureSubscription: ${{ variables.armServiceConnectionName }}
+      scriptType: bash
+      scriptLocation: inlineScript
+      inlineScript: |
+        cd appmonitoring/scripts
+        pwd
+        az account set --subscription 66010356-d8a5-42d3-8593-6aaa3aeb1c11
+        az aks get-credentials --resource-group appmonitoring-pipeline-test --name aks-pipeline-testbed-ai
+
+        sudo chmod u+x ./validate-aks-charts.sh
+        sudo chmod u+x ./validate-certs.sh
+
+        echo "Checking for charts..."
+        ./validate-aks-charts.sh
+        
+        echo "Checking for certificate installation..."
+        ./validate-certs.sh
+
+  # - task: AzureResourceManagerTemplateDeployment@3
+  #   inputs:
+  #     deploymentScope: 'Resource Group'
+  #     azureResourceManagerConnection: 'ai-acr-validation-rsx'
+  #     subscriptionId: '66010356-d8a5-42d3-8593-6aaa3aeb1c11'
+  #     action: 'Create Or Update Resource Group'
+  #     resourceGroupName: 'appmonitoring-pipeline-test'
+  #     location: 'East US'
+  #     templateLocation: 'Linked artifact'
+  #     csmFile: '/path/to/here'
+  #     deploymentMode: 'Incremental'
+
+  - task: AzureCLI@2
+    displayName: "Install CR and test apps"
+    inputs:
+      azureSubscription: ${{ variables.armServiceConnectionName }}
+      scriptType: bash
+      scriptLocation: inlineScript
+      inlineScript: |
+        cd appmonitoring/helm
+        pwd
+        az account set --subscription 66010356-d8a5-42d3-8593-6aaa3aeb1c11
+        az aks get-credentials --resource-group appmonitoring-pipeline-test --name aks-pipeline-testbed-ai
+        kubectl delete -f java-test-app.yaml
+        kubectl delete -f dotnet-test-app.yaml
+        kubectl delete -f node-test-app.yaml
+        kubectl delete -f appmonitoring-cr.yaml
+
+        echo "Wait for 10s for everything to clear up..."
+        sleep 10
+
+        export JAVA_TEST_IMAGE_NAME=${{ variables.javaTestAppImageName }}
+        export DOTNET_TEST_IMAGE_NAME=${{ variables.dotnetTestAppImageName }}
+        export NODEJS_TEST_IMAGE_NAME=${{ variables.nodeTestAppImageName }}
+        export TEST_NS="${{ variables.testNamespace }}"
+        export AI_CONNECTION_STRING="${{ variables.aiConnectionString }}"
+
+        kubectl create namespace $TEST_NS
```

Comment:
Are we creating a new namespace for every test? Are we leaving all the old ones behind?

Reply:
no, its the same and we delete all that came before.

Reply:
no, its the same. We delete all resources from previous run

Reply:
Won't work for concurrent pipelines. We need to either figure out how to make sure different runs don't affect each other, or how to prohibit and fail them with a meaningful message. Are all our resources namespace-scoped or not? Can we support multiple runs at once?

Reply:
we cannot do concurrent tests, as many of the things being created here can only exist one at a time in a cluster, like webhook, CRD, CR etc. I will restrict the pipeline to only running it one at a time

Reply:
If you can restrict the pipeline itself somehow - that would work. If not, then let's make sure we check in the scripts and bail out if the environment is used.

Another option would be to make those cluster-scoped objects per-run, like add the id of the run to the end of the object names... any way we could do that? It would be pretty awesome to support multiple runs in parallel.